### PR TITLE
[doc/security] Add Device Life Cycle Specification

### PR DESCRIPTION
**Pull request by moidx**
This commit converts to markdown the Device Life Cycle specification
referenced in RFC lowrisc/opentitan#1294.

This document describes the OpenTitan device operational states supported for
manufacturing, provisioning, regular operation and return to
manufaturer.

The document was converted to markdown by a GDoc plugin and formatted by
a md-linter. Since the maximum line width is set to 80cols, in most
cases a sentence change will trigger a full paragraph reformat/diff.

The GDoc to Markdown pluging generates tables in HTML format, which in
this case is useful to retain the per-cell text formatting and the rows
and columns layout.

Signed-off-by: Miguel Osorio <miguelosorio@google.com>

### CODE REVIEW

**Comment by moidx** [comment]
Hi @cdgori, @asb, @sjgitty, @gkelly, please let me know if you have any comments on this PR. Thanks!

File: doc/security/specs/device_life_cycle/_index.md [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+cannot be exited once entered.
+
+OWNED / UNOWNED identity states are defined in flash and thus can be reversed
+under the right privileges.
+
+Below, each state is described in detail.
+
+### Manufacturing States
+
+<table>
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md:103]: The table was initially generated in HTML format by the export tool.  I looked at formatting it in markdown, but the issue I run into is that these tables have additional formatting at the cell level, and some have merged columns or rows. 

At the end it html ended up formatting the html for readability given the limitations above.  Are you ok if we keep this in HTML for now?

**Comment by asb** [comment]
@rbohn - you mentioned you had some editorial/presentational feedback when we reviewed this in the Technical Committee. Are there any clarifications you'd like to propose prior to this initial version being committed?

File: doc/security/specs/device_life_cycle/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rswarbrick) [reply doc/security/specs/device_life_cycle/_index.md:103]: As a note: one thing you can do is to put markdown formatting inside an HTML table, so long as you separate it from the HTML tags with blank lines. So something like:
```
<table><tr><td>

`this is` **markdown**

 with | a
--|--
 nested | table

</td><td>

And this is to the right

</td></tr></table>
```

I used this trick in the OTBN document generator (which wanted to squirt hunks of markdown into a "complicated" table), and it seems to have worked quite nicely.

Probably no change needed to this PR, but if you come back and want to tidy up the formatting in the future, you could probably use this to get rid of some of the HTML tags.

File: doc/security/specs/device_life_cycle/_index.md [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+        <li>CPU execution is enabled
+        <li>NVM backdoor functions are enabled
+        <li>Debug functions are enabled
+        <li>DFT functions are enabled</li>
+      </ul>
+      <p>
+      Note: during this state it is not possible to provision specific OTP root
+      secrets. This will be enforced by hardware. If this dependency is
+      violated, DFT, CPU and debug functionality will be revoked until the next
+      power cycle.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply doc/security/specs/device_life_cycle/_index.md:155]: Yeah i think the line beginning at `If this dependency` is somewhat outdated and I forgot to remove it.

File: doc/security/specs/device_life_cycle/_index.md [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+attributes defined by each that are not strictly related most of the time. By
+decoupling a single definition into two, more flexibility is allowed in the
+system.
+
+In addition, while certain states are defined by OTP (thus hard), other states
+are defined by flash. This gives the latter more flexibility in changing values
+to support the provisioning flow ownership transfer model.
+
+For example, DEV / PROD / RMA manufacturing states are defined in OTP states and
+cannot be exited once entered.
```
Comment (rbohn) [inline doc/security/specs/device_life_cycle/_index.md:94]: Is this line accurate? The way I read it implies that neither DEV nor PROD can be exited once entered. But the diagram shows exist transitions into the RMA state. Then more pedantically, even the RMA state has an exit condition to both the SCRAP and INVALID states. 

It looks like the overall point you are trying to make is less about the states, but more about the directionality of the transitions. You can go between OWNED / UNOWNED repeatedly, but you can't go from RMA back to PROD, or PROD back to TEST. 

"For example, transitions into the DEV / PROD / RMA manufacturing states affect OTP states and therefore cannot be reversed once completed."

File: doc/security/specs/device_life_cycle/_index.md [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+transitions cannot be reverted, while flash defined states are meant to be
+'flexible', especially when ownership transfer is required. This implies that
+when OTP states are transitioned from A to B, we cannot go back to A from B.
+However, it is still possible to transition from B to C, assuming such a
+transition is legal and allowed by the life cycle definition.
+
+See the table below for more details.
+
+<img src="device_life_cycle_fig1.svg" alt="Fig1" style="width: 800px;"/>
+
```
Comment (rbohn) [inline doc/security/specs/device_life_cycle/_index.md:58]: I noticed that the figure shows all of the states mentioned in the tables below except for the INVALID state. Should the INVALID state be added to this diagram / similar to the SCRAP state shown? Entry ARC might be something like "any invalid OTP state."

File: doc/security/specs/device_life_cycle/_index.md (hunk 3 above)
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: SGMT. Updated wording. Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+        When transitioning from TEST, no RMA_UNLOCK token is required.
+      <p>
+        A hardware-backed mechanism will erase all owner flash content before
+        RMA transition is allowed. This includes the isolation owner flash
+        partitions as well as any owner code. Once erasure is confirmed and
+        verified, RMA transition will proceed.
+      <p>
+        During this state
+      <ul>
+        <li>CPU execution is enabled
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+   <td>Transition from TEST state via OTP write. This is a mutually exclusive
+       state to DEV and PROD_END.
+     <p>
+       To enter this state, a TEST_EXIT token is required.
+     <p>
+       This state is used for developing provisioning and mission mode software.
+     <p>
+       During this state
+     <ul>
+       <li>CPU execution is enabled
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Switched to markdown.

Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 7]
```diff
@@ -0,0 +1,646 @@
...
+        <li>NVM backdoor functions are disabled
+        <li>Debug functions are disabled
+        <li>DFT functions are disabled</li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <td>PROD_END</td>
+    <td>OTP</td>
+   <td>This state is identical in functionality to PROD, except the device is
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Fixed.

File: doc/security/specs/device_life_cycle/_index.md [hunk 8]
```diff
@@ -0,0 +1,646 @@
...
+    <td>Transition from TEST state via OTP write. This is a mutually exclusive
+        state to DEV and PROD_END.
+      <p>
+        To enter this state, a TEST_EXIT token is required.
+      <p>
+        This state is used both for provisioning and mission mode.
+      <p>
+        During this state:
+      <ul>
+        <li>CPU execution is enabled
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Switched to markdown. Done.

File: doc/security/specs/device_life_cycle/_index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Removed, and Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 9]
```diff
@@ -0,0 +1,646 @@
...
+    <td>TEST{N}</td>
+    <td>OTP</td>
+    <td>
+      Transition from RAW state using OTP write.
+      <p>
+      This state is used for <strong><em>manufacturing and production testing.</em></strong>
+      <p>
+      During this state:
+      <ul>
+        <li>CPU execution is enabled
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Switched to markdown. 

Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 10]
```diff
@@ -0,0 +1,646 @@
...
+      token is required.
+    </td>
+  </tr>
+  <tr>
+    <td>TEST{N}</td>
+    <td>OTP</td>
+    <td>
+      Transition from RAW state using OTP write.
+      <p>
+      This state is used for <strong><em>manufacturing and production testing.</em></strong>
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: The parser seemed unable to handle the formatting suggestion inside the table... need to debug. Ok TODO?

File: doc/security/specs/device_life_cycle/_index.md [hunk 11]
```diff
@@ -0,0 +1,646 @@
...
+      <p>
+      The token authorizing the transition from RAW to TEST0 is a value that
+      is secret global to all devices.  This is known as the RAW_UNLOCK token.
+    </td>
+  </tr>
+  <tr>
+    <td>TEST{N}_LOCKED</td>
+    <td>OTP</td>
+    <td>
+      TEST{N}_LOCKED states have identical functionality as RAW and serve as a
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 12]
```diff
@@ -0,0 +1,646 @@
...
+
+In addition, while certain states are defined by OTP (thus hard), other states
+are defined by flash. This gives the latter more flexibility in changing values
+to support the provisioning flow ownership transfer model.
+
+For example, DEV / PROD / RMA manufacturing states are defined in OTP states and
+cannot be exited once entered.
+
+OWNED / UNOWNED identity states are defined in flash and thus can be reversed
+under the right privileges.
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Done.

File: doc/security/specs/device_life_cycle/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md:103]: Updated the inner cell formatting to follow markdown as suggested by @rswarbrick.

Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 13]
```diff
@@ -0,0 +1,646 @@
...
+Manufacturing refers to the process in which a device is tested and prepared for
+silicon correctness prior to production.
+
+## Background
+
+First see [here]({{< relref "doc/security/logical_security_model" >}}) for
+OpenTitan's logical entity breakdowns.
+
+OpenTitan implements standard device life cycle manufacturing states which are
+configured via OTP[^2] memory. These states allow the **silicon creator** to
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: Done.

File: doc/security/specs/device_life_cycle/_index.md [hunk 14]
```diff
@@ -0,0 +1,646 @@
...
+
+<p style="text-align: right">
+    Version: 0.1</p>
+
+<p style="text-align: right">
+  Contributors:
+  <a href="http://github.com/tjaychen">@tjaychen</a>,
+  <a href="http://github.com/moidx">@moidx</a>,
+  <a href="http://github.com/ccgori">@ccgori</a>
+</p>
```
[the remark this answers is not in the document]
Comment (moidx) [reply doc/security/specs/device_life_cycle/_index.md]: SGTM. I will look into adding this in a follow up change.

**Comment by moidx** [comment]
Updated PR with requested changes.

**Pull request merged**